### ISSUE TITLE
Performance: one training epoch exceeds 60 s on nv0 for XML text dataset with attention and MoE model

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#465]
The reproduction uses `recipe.data("data/text/records_xml.xml").target("target")` (12,000 rows, text feature width ~95 bytes) and a model with `attn(1).relu().iq(2).m`, `layer(8)`, `moe(1, [layer(8), layer(8)]).tanh().norm(batch).iq(3).xxs`, trained with `recipe.train().optimizer(adamw).lr(0.0001).seed(38004140618274).epochs(1).bf(16).run()`. The measured epoch time exceeds the 60-second slow-runtime threshold (`elapsed_seconds:60 backend:nv0`, stable:true). The workload (attention QKV, dense projection, two expert branches, batch normalization, quantized contraction nodes, 12k rows) does not justify >60 s per epoch on the available GPU, so the measured path is materially slower than the workload and hardware justify. No existing issue covers this earliest training-time failure.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `748682736fd25811`
- Classifier provider: `openrouter`
- Classifier model: `thinkingmachines/inkling:free`
- Classifier effort: `default`

### Classifier decision

```json
{
  "verdict": "new",
  "issue": 0,
  "title": "Performance: one training epoch exceeds 60 s on nv0 for XML text dataset with attention and MoE model",
  "body": "The reproduction uses `recipe.data(\"data/text/records_xml.xml\").target(\"target\")` (12,000 rows, text feature width ~95 bytes) and a model with `attn(1).relu().iq(2).m`, `layer(8)`, `moe(1, [layer(8), layer(8)]).tanh().norm(batch).iq(3).xxs`, trained with `recipe.train().optimizer(adamw).lr(0.0001).seed(38004140618274).epochs(1).bf(16).run()`. The measured epoch time exceeds the 60-second slow-runtime threshold (`elapsed_seconds:60 backend:nv0`, stable:true). The workload (attention QKV, dense projection, two expert branches, batch normalization, quantized contraction nodes, 12k rows) does not justify >60 s per epoch on the available GPU, so the measured path is materially slower than the workload and hardware justify. No existing issue covers this earliest training-time failure.",
  "rationale": "Search returned no matching issues (`performance`, `slow`, `runtime`, `epoch`, `training` all empty). The failure is a deterministic performance packet (`kind=performance`, `replay` stable). The earliest observed failure is the training epoch exceeding the configured slow-runtime threshold (`recipe.train()` on the given data/model). The inspected source shows the data loader creates ~12,000 text-feature rows (~95-byte width) and the model lowers to attention, dense layers, MoE with two layer(8) experts, normalization, and quantized nodes; this arithmetic and memory footprint should complete well within 60 s on `nv0`. The same earliest failing operation (training epoch) and same implementation path (public `recipe.train()` running the compiled graph) are not represented by any existing issue, so this is a new performance report rather than additional evidence for an existing loader or model issue."
}
```

## Deterministic failure packet

kind=performance
id=748682736fd25811
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=clean
cursor=seed:0 cursor:18006 next:18007 composition:38004140618274
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:nv0
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-2634-18006/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/text/records_xml.xml")
		.target("target");
	let model = recipe.model()
		.attn(1).relu().iq(2).m
		.layer(8)
		.moe(1, [layer(8), layer(8)]).tanh().norm(batch).iq(3).xxs
		.loss(mae);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(38004140618274)
		.epochs(1)
		.log(all)
		.bf(16)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=nv0

**Comment by nm-z** [comment]
Closing as resolved based on zero-change verification PR #527.

That PR changes 0 files and verifies the existing `minimal` behavior, so there is no code to merge. The original verification is copied below.

<details>
<summary>Verification from PR #527: Verify XML attention and MoE runtime</summary>

## Summary

- Reproduce the deterministic XML text, attention, and MoE performance packet on current minimal.
- Confirm that the public training path stays far below the reported 60-second threshold.

## Verification

- The training epoch took 317.864 ms.
- The complete public run took 11.660 seconds and saved a 1,349,154-byte bundle.
- Public inference loaded the saved bundle and returned finite output `0.5` in 3.234 seconds.
- Archy finished with 22 GiB available, zero swap use, and all six GPUs idle at 27 to 35°C.

Fixes #465

</details>

Accepted verdict: PASS

Reason: The verification above reports that current `minimal` completes this issue's reported path successfully.
